### DISCUSSION TITLE
What buffer size would be best for streaming using SDK?

### DISCUSSION THREAD

**Discussion by LeeKevinXO** [deepgram/community#224] [General help] [answered]
I tried 100ms, 200ms buffer size that means sending 100ns/200ms data each time to the server.
But each result may lost some words at the beging or the ending.

**Comment by jjmaldonis** [comment, marked as answer]
Hey @LeeKevinXO, we recommend sending between 20ms and 250ms of data at once, so 200ms should work well.

If a word gets cut off in the middle of an audio segmentation, the word likely won't get transcribed. Is that what you are seeing? If so, the transcript will update previous results as more audio comes in, so the skipped words should get re-added after a few seconds.

**Reply by jjmaldonis** [reply]
@LeeKevinXO an update: the transcript's previous results will only get updated if `interim_results=true`, so that may be a parameter to test out.

We'll make an update to our documentation in the next day or two with this additional info:
> - **Buffer size**: The amount of audio sent in each streaming message. Too large of a buffer adds built-in delay to getting results while the audio buffers. Too small of a buffer adds a lot of tiny packets, degrading network performance. Streaming buffer sizes should be between 20 milliseconds and 250 milliseconds of audio, with 100 milliseconds often striking a god balance.
